html/nso/sbo/index.js

```diff
@@ -101,7 +106,9 @@ function logout() {
 }
 
 function createTab(title, tabId) {
-	if (typeof title == "string") title = $("<a>").html(title);
+	if (typeof title == "string") title = $("<a>").html(title).click(function() {
+		location.hash = '#'+tabId;
```

Comment:
This isn't working with nested tabs. I'd suggest removing up/download from the operator screen (they can do logos from the teams page now anyway and that's the thing changed most often). I'd also put rulesets last rather than 2nd, it's pretty rare that they need changing.

Reply:
> This isn't working with nested tabs.

Neither would a separate browser tab.

> I'd suggest removing up/download from the operator screen (they can do logos from the teams page now anyway and that's the thing changed most often).

I'd anticipate that downloading JSON files will become regularly used. (Plug in USB key, go to download tab and save to the key - no need to find the correct folder via explorer etc)

> I'd also put rulesets last rather than 2nd, it's pretty rare that they need changing.

Usage frequency has never been a factor in the tab order (or Teams would be 2nd). This is the place where historically the Policies tab has been, which has been replaced by the rulesets. So it's one less item that people skipping several versions will have to adjust to.

Reply:
> Neither would a separate browser tab.

It'd work fine by default.

> I'd anticipate that downloading JSON files will become regularly used.

By the operator during a game though?

> Usage frequency has never been a factor in the tab order (or Teams would be 2nd).

Maybe it's time to switch that then?

Reply:
> It'd work fine by default.

How could it? It'd jump to the first (sub)tab just as the operator screen did all the time.

> By the operator during a game though?

Right after the game or between games of a double or triple header seems likely.

> Maybe it's time to switch that then?

If you want to. I suggest that you start by collecting representative usage stats.
But I'm not willing to have this PR sit there and collect more merge conflicts in the meantime.

Reply:
> How could it? It'd jump to the first (sub)tab just as the operator screen did all the time.

If it's a browser tab, then refresh leaves you where you were.

> Right after the game or between games of a double or triple header seems likely.

By a game data manager or stats crew maybe, but not the operator.

Reply:
> If it's a browser tab, then refresh leaves you where you were.

Yeah, sure. That's why I had to add code so that the operator screen remembers its opened tab on refresh.

> By a game data manager or stats crew maybe, but not the operator.

HNSO or home league rep come to the SBO and ask: "Can you quickly put the game data file on this stick so I can generate the statsbook at home?" Or they do it themselves but that doesn't change the fact that it is more convenient to be able to do it from the screen that's already open than having to open another one.

Reply:
> That's why I had to add code so that the operator screen remembers its opened tab on refresh.

That doesn't work for the nested tabs on the Up/Download on the operator screen.

I'd suggest removing the Up/Download screen (and Rulesets), as use of them will be relatively niche and resolves this issue. In such a setup the HNSO likely has network access anyway, and can pull it themselves.

Reply:
> That doesn't work for the nested tabs on the Up/Download on the operator screen.

We're going in circles here - I never claimed it would. You requested it to remember the opened tab on reload as it would do with separate browser tabs. I implemented that. If you want more features, I suggest you open a feature request in the issue tracker instead of trying to continuously scope creep my PR.

Reply:
It doesn't do that though, which your original proposal would have. The creep is due to changing from the original proposal.

Reply:
Ok, I've reverted the behaviour to current HEAD. If you want any changes to how tabs behave, bring those up separate from this PR.